src/libpriv/rpmostree-scripts.c

```diff
@@ -213,14 +212,75 @@ run_script_in_bwrap_container (int rootfs_fd,
   return ret;
 }
 
+/* Medium level script entrypoint; we already validated it exists and isn't
+ * ignored. Here we mostly compute arguments/input, then proceed into the lower
+ * level bwrap execution.
+ */
 static gboolean
-run_known_rpm_script (const KnownRpmScriptKind *rpmscript,
-                      DnfPackage    *pkg,
-                      Header         hdr,
-                      GHashTable    *ignore_scripts,
-                      int            rootfs_fd,
-                      GCancellable  *cancellable,
-                      GError       **error)
+impl_run_rpm_script (const KnownRpmScriptKind *rpmscript,
+                     DnfPackage    *pkg,
+                     Header         hdr,
+                     int            rootfs_fd,
+                     GCancellable  *cancellable,
+                     GError       **error)
+{
+  const char *script = headerGetString (hdr, rpmscript->tag);
+  g_assert (script);
+
+  struct rpmtd_s td;
+  g_autofree char **args = NULL;
+  if (headerGet (hdr, rpmscript->progtag, &td, (HEADERGET_ALLOC|HEADERGET_ARGV)))
+    args = td.data;
+
+  const char *interp = args && args[0] ? args[0] : "/bin/sh";
```

Comment:
This isn't new in this patch, though while we're here, can we wrap the `&&` operands in parentheses?